src/front/wgsl/mod.rs

```diff
@@ -1257,6 +1257,72 @@ impl Parser {
             }
         } else {
             match name {
+                "bitcast" => {
+                    let _ = lexer.next();
+                    lexer.expect_generic_paren('<')?;
+                    let ((ty, _access), type_span) = lexer.capture_span(|lexer| {
+                        self.parse_type_decl(lexer, None, ctx.types, ctx.constants)
+                    })?;
+                    lexer.expect_generic_paren('>')?;
+
+                    let expr = self.parse_general_expression(lexer, ctx.reborrow())?;
+
+                    // Create error that will be returned if type matching fails
+                    let err = Err(Error::BadTypeCast {
+                        from_type: format!("{:?}", ctx.resolve_type(expr)?),
+                        span: type_span,
+                        to_type: format!("{:?}", ctx.types[ty].inner),
+                    });
+
+                    match ctx.types[ty].inner {
+                        crate::TypeInner::Scalar { kind, .. } => match *ctx.resolve_type(expr)? {
+                            crate::TypeInner::Scalar { .. } => crate::Expression::As {
```

Comment:
We don't need to validate here at all, leave it to IR validator.
If we know this is `bitcast`, all we really need is the `kind`, which is extracted from the type. There should be no need to match any expression types here

Reply:
OK, cool, that makes it simpler.

You always come at my pull requests with like a PR shrink ray. :smile: